puma/hlplots/tagger.py

```diff
@@ -0,0 +1,214 @@
+"""Tagger module for high level API."""
+import h5py
+import numpy as np
+import pandas as pd
+from numpy.lib.recfunctions import structured_to_unstructured
+
+from puma.utils import calc_disc, logger
+
+
+class TaggerBase:  # pylint: disable=too-many-instance-attributes
+    """Base class handling tagger results."""
+
+    def __init__(self, model_name: str) -> None:
+        """Init Tagger class.
+
+        Parameters
+        ----------
+        model_name : str
+            Name of the model, also correspondinng to the pre-fix of the tagger
+            variables.
+        """
+        self.model_name = model_name
+        self.label = None
+        self.reference = False
+
+        self.scores = None
+        self.perf_var = None
+        self.flvs = ["pu", "pc", "pb"]
+
+        self.is_b = None
+        self.is_light = None
+        self.is_c = None
+
+    def extract_tagger_scores(
+        self, source: object, source_type: str = "data_frame", key: str = None
```

Comment:
rather than a single `source` which cannot be type hinted and a `source_type`, I would suggest to have `df : DataFrame = None` and `path: Path = None` arguments and the logic can be based on which of the two is defined.

Reply:
how would we distinguish then h5 with structured numpy or with a panda data frame ?

Reply:
hmm, we could just trivially convert structured numpy to dataframe if provided? or ask the user to do it upstream?

Reply:
let's move that also into a follow up issue